Users/Tutorial.rst

```diff
@@ -264,24 +267,32 @@ That looks horrible, doesn't it? Let's fix it!
     $ coala -S python.bears=PEP8Bear python.files=\*\*/\*.py \
     --apply-patches --save
     # other output ...
+    Executing section cli...
     Executing section python...
     [INFO][11:03:37] Applied 'ApplyPatchAction' for 'PEP8Bear'.
     [INFO][11:03:37] Applied 'ApplyPatchAction' for 'PEP8Bear'.
 
 coala would now fix all spacing issues and without bothering you again.
 
+.. note::
+
+    When you try the above example, you may get a warning, saying that all
+    settings in the ``cli`` section are implicitly inherited to all
+    other sections (if they do not override their values). It also advises
+    us to change the name of that section to avoid unexpected behavior.
+
 Setting Inheritance
 -------------------
+Let's first see what inheritance means.
 
-All settings in the default section are implicitly inherited to all
-other sections (if they do not override their values). We can use that
-to save a few lines!
+Before proceeding, rename the ``cli`` section in the ``.coafile`` to
+``all`` (we will soon explain the reason behind this change).
 
-Lets add the following section to our ``.coafile``:
+Now, add the following section to our ``.coafile``:
 
 ::
 
-    [TODOS]
+    [all.TODOS]
```

Comment:
Hm but why `all` in fact? For this example we could keep `TODOS` itself, the reader wouldn't understand the usage of `all` here.

Reply:
If we keep it as just `TODOS` then we will be demonstrating the implicit inheriting of the `cli` section using an example. IMHO we should not encourage this behaviour. But we still need to demonstrate the inheritance feature-- enter `all`. :wink:

Reply:
setting inheritance is explained completely only in the doc for writing configuration files, usage of inheritance from all is left unexplained in here

Reply:
I think the setting inheritance section would be best removed from here or put the link to the doc for writing config files here

Reply:
Yep agreed! :-) 
Should open a new issue for this maybe because I think that it will be out of the scope of the the current issue?

Reply:
since you are changing many things, you can mention this on the issue, the commit body and pull it off here. Anyways your call 😄.

Reply:
I really think that removing the **Setting Inheritance** should go to a new issue. That is if that's okay with you? :-)